### PR TITLE
Improve editor

### PR DESCRIPTION
**Pull request by ArsalanSavand**
### Massive update for editor:
- Added a button that will embed YouTube videos.
- Added a method to add embed YouTube videos when pasting their raw link into editor.
- Now editor is full-screen.
- Changed all editor icons from SVGs to FontAwesome.
- Sidebar is no longer available when writing.

### Horizontal line
image

### Final ui
image

### CODE REVIEW

**Comment by Alir3z4** [comment]
Delicious 😋
This is so good.

I'll review this first thing in the morning.

**Comment by Alir3z4** [comment]
Good work on the editor.
The divider looks good as well.

Some change requests:

*  The editor takes a lot of of unnecessary space with its height, reduce it as much as it needs. Let its height to be automatic or something that doesn't make this tick.
* the divider doesn't show up the way it is in the editor. They should look similar in both backend and editor.  image
* Youtube videos are very small, there is no size on them, therefore when inserted they become very small. This can be fixed with passing style on the iframe node `style="width: -webkit-fill-available;height:  -webkit-fill-available;` but I'm not sure about its compatibility. Figure out a solution to properly handle the size of the embedded video.
* What happened to Publish changes button ? Now it's very confusing, I only see the "Update" button and when pressed it will publish the unpublished changes. Please bring back the previous flow where user could actually know he is either saving the changes or publishing them.

**Comment by ArsalanSavand** [comment]
@Alir3z4  these are the answers to your questions above:
I removed unwanted spaced inside editor buttons.
The divider is just a style, we have to also copy that style in Gonevis.
I have finally added a method to make every YouTube videos 16:9. ( also full-screen ).
Publish button exists, try again and see.

File: src/app/entry/entry_edit/entry_edit_controller.js [hunk 1]
```diff
@@ -230,6 +260,24 @@ function EntryEditController($scope, $rootScope, $state, $stateParams, $timeout,
       delta.ops = ops;
       return delta;
     });
+
+    editor.clipboard.addMatcher(Node.TEXT_NODE, function(node, delta) {
+      let regex = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=|\?v=)([^#\&\?]*).*/;
```
[the remark this answers is not in the document]
Comment (ArsalanSavand) [reply src/app/entry/entry_edit/entry_edit_controller.js]: That also works.

File: src/app/entry/entry_edit/editor.js [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
+  icon: "clean",
+  replace: "ban"
+}, {
+  icon: "publish",
+  replace: "globe",
+  text: "Publish"
+}, {
+  icon: "update",
+  replace: "refresh",
+  text: "Update"
```
[the remark this answers is not in the document]
Comment (ArsalanSavand) [reply src/app/entry/entry_edit/editor.js]: I have no Idea, but I will something happen don't worry.

[the remark this answers is not in the document]
Comment (AmirSavand) [reply src/app/entry/entry_edit/editor.js]: No, this needs to be translated.

File: src/app/entry/entry_edit/entry_edit_controller.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AmirSavand) [reply src/app/entry/entry_edit/entry_edit_controller.js]: Did you write a test for it? I don't think this will work, write tests for it.

**Pull request merged**